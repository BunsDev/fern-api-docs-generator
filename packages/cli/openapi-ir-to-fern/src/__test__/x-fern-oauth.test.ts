import { testConvertOpenAPI } from "./testConvertOpenApi";

describe("x-fern-oauth", () => {
    testConvertOpenAPI("x-fern-oauth", "openapi.yml");
});
