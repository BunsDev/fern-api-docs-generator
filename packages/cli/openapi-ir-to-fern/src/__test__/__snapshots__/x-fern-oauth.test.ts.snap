// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`x-fern-oauth x-fern-oauth docs 1`] = `
{
  "definitionFiles": {},
  "packageMarkerFile": {
    "service": {
      "auth": false,
      "base-path": "",
      "endpoints": {
        "postRefresh": {
          "auth": false,
          "docs": undefined,
          "examples": [
            {
              "query-parameters": {
                "refresh_token": "refresh_token",
              },
              "response": {
                "body": {
                  "accessToken": "accessToken",
                  "refreshToken": "refreshToken",
                },
              },
            },
          ],
          "method": "POST",
          "pagination": undefined,
          "path": "/refresh",
          "request": {
            "name": "PostRefreshRequest",
            "query-parameters": {
              "refresh_token": "string",
            },
          },
          "response": {
            "docs": "A real life token",
            "type": "PostRefreshResponse",
          },
        },
        "postToken": {
          "auth": false,
          "docs": undefined,
          "examples": [
            {
              "query-parameters": {
                "client_id": "client_id",
                "client_secret": "client_secret",
                "scopes": "scopes",
              },
              "response": {
                "body": {
                  "accessToken": "accessToken",
                  "refreshToken": "refreshToken",
                },
              },
            },
          ],
          "method": "POST",
          "pagination": undefined,
          "path": "/token",
          "request": {
            "name": "PostTokenRequest",
            "query-parameters": {
              "client_id": "string",
              "client_secret": "string",
              "scopes": "string",
            },
          },
          "response": {
            "docs": "A real life token",
            "type": "PostTokenResponse",
          },
        },
      },
    },
    "types": {
      "PostRefreshResponse": {
        "docs": undefined,
        "properties": {
          "accessToken": "string",
          "refreshToken": "string",
        },
      },
      "PostTokenResponse": {
        "docs": undefined,
        "properties": {
          "accessToken": "string",
          "refreshToken": "string",
        },
      },
    },
  },
  "rootApiFile": {
    "auth-schemes": {
      "OAuthScheme": {
        "client-id-env": "MY_CLIENT_ID",
        "client-secret-env": "MY_CLIENT_ID",
        "get-token": {
          "endpoint": "root.postToken",
          "request-properties": {
            "client-id": "$query.client_id",
            "client-secret": "$query.client_secret",
            "scopes": "$query.scopes",
          },
          "response-properties": {
            "access-token": "$response.access_token",
            "expires-in": undefined,
            "refresh-token": "$response.refresh_token",
          },
        },
        "refresh-token": {
          "endpoint": "root.postRefresh",
          "request-properties": {
            "refresh-token": "$query.refresh_token",
          },
          "response-properties": {
            "access-token": "$response.access_token",
            "expires-in": undefined,
            "refresh-token": "$response.refresh_token",
          },
        },
        "scheme": "oauth",
        "scopes": [
          "scope1",
          "scope2",
        ],
        "token-prefix": "Bearer",
        "type": "client-credentials",
      },
    },
    "display-name": "Test extension \`x-fern-oauth\`.",
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;

exports[`x-fern-oauth x-fern-oauth simple 1`] = `
{
  "definitionFiles": {},
  "packageMarkerFile": {
    "service": {
      "auth": false,
      "base-path": "",
      "endpoints": {
        "postRefresh": {
          "auth": false,
          "docs": undefined,
          "examples": [
            {
              "query-parameters": {
                "refresh_token": "refresh_token",
              },
              "response": {
                "body": {
                  "accessToken": "accessToken",
                  "refreshToken": "refreshToken",
                },
              },
            },
          ],
          "method": "POST",
          "pagination": undefined,
          "path": "/refresh",
          "request": {
            "name": "PostRefreshRequest",
            "query-parameters": {
              "refresh_token": "string",
            },
          },
          "response": {
            "docs": "A real life token",
            "type": "PostRefreshResponse",
          },
        },
        "postToken": {
          "auth": false,
          "docs": undefined,
          "examples": [
            {
              "query-parameters": {
                "client_id": "client_id",
                "client_secret": "client_secret",
                "scopes": "scopes",
              },
              "response": {
                "body": {
                  "accessToken": "accessToken",
                  "refreshToken": "refreshToken",
                },
              },
            },
          ],
          "method": "POST",
          "pagination": undefined,
          "path": "/token",
          "request": {
            "name": "PostTokenRequest",
            "query-parameters": {
              "client_id": "string",
              "client_secret": "string",
              "scopes": "string",
            },
          },
          "response": {
            "docs": "A real life token",
            "type": "PostTokenResponse",
          },
        },
      },
    },
    "types": {
      "PostRefreshResponse": {
        "docs": undefined,
        "properties": {
          "accessToken": "string",
          "refreshToken": "string",
        },
      },
      "PostTokenResponse": {
        "docs": undefined,
        "properties": {
          "accessToken": "string",
          "refreshToken": "string",
        },
      },
    },
  },
  "rootApiFile": {
    "auth-schemes": {
      "OAuthScheme": {
        "client-id-env": "MY_CLIENT_ID",
        "client-secret-env": "MY_CLIENT_ID",
        "get-token": {
          "endpoint": "root.postToken",
          "request-properties": {
            "client-id": "$query.client_id",
            "client-secret": "$query.client_secret",
            "scopes": "$query.scopes",
          },
          "response-properties": {
            "access-token": "$response.access_token",
            "expires-in": undefined,
            "refresh-token": "$response.refresh_token",
          },
        },
        "refresh-token": {
          "endpoint": "root.postRefresh",
          "request-properties": {
            "refresh-token": "$query.refresh_token",
          },
          "response-properties": {
            "access-token": "$response.access_token",
            "expires-in": undefined,
            "refresh-token": "$response.refresh_token",
          },
        },
        "scheme": "oauth",
        "scopes": [
          "scope1",
          "scope2",
        ],
        "token-prefix": "Bearer",
        "type": "client-credentials",
      },
    },
    "display-name": "Test extension \`x-fern-oauth\`.",
    "error-discrimination": {
      "strategy": "status-code",
    },
    "name": "api",
  },
}
`;
